sw/airborne/modules/gps/gps_ubx_ucenter.c

```diff
@@ -167,10 +167,10 @@ void gps_ubx_ucenter_event(void)
   // Read Configuration Reply's
   switch (gps_ubx.msg_class) {
     case UBX_ACK_ID:
-      if (gps_ubx.msg_id & UBX_ACK_ACK_ID) {
+      if (gps_ubx.msg_id == UBX_ACK_ACK_ID) {
```

Comment:
See a few lines below - gps_ubx.msg_id == xxx is used instead of &
UBX_ACK_ACK_ID = 0x01, UBX_ACK_NAK_ID = 0x00
Before the second comparison was always false, now both should work fine.
Btw the whole module is poorly written..:(